theano/gof/wrapper.py

```diff
@@ -101,7 +101,7 @@ def __init__(self, wrapper, **kwargs):
             if field not in kwargs:
                 raise TypeError('Wrap: Wrapper attribute "%s" not in Wrap args.' % field)
         super(Wrap, self).__init__(**kwargs)
-        self.__dict__.update(wrapper=wrapper)
+        self.__dict__.update(wrapper=wrapper, signatures=None)
```

Comment:
Use \__signatures__ to lower the probability of having that name used by the user.

if there is a colision, there will be an error already.